From 24bfbcdf341b7f1bb760cf80af1d57a347e20cc3 Mon Sep 17 00:00:00 2001
From: Karthikeyan K <karthikk@habana.ai>
Date: Thu, 03 Feb 2022 08:07:58 +0200
Subject: [PATCH] [SW-68803] boot_desc: update new fields from specs

Image pointers have been updated with new fields, namely fw version,
public key size and signature size. This needs to be updated also in
the internal boot descriptor structure which uses these fields.

Change-Id: I165e04becf79a6fd9958187799cfcf022f437507
---

diff --git a/common/boot_desc.c b/common/boot_desc.c
index 15aa11b..c4bbf70 100644
--- a/common/boot_desc.c
+++ b/common/boot_desc.c
@@ -148,6 +148,9 @@
 			p_img_desc[i].load_high[idx] << 32) |
 			p_img_desc[i].load_low[idx];
 		desc->desc_ver = p_img_desc[i].desc_ver;
+		desc->sec_ver = p_img_desc[i].sec_ver[idx];
+		desc->pub_key_size = p_img_desc[i].pub_key_size[idx];
+		desc->sig_size = p_img_desc[i].sig_size[idx];
 
 		if (img != HL_FW_BOOT_PTR_SPI)
 			memcpy(desc->chksum, p_img_desc[i].chksum2[idx],
diff --git a/common/boot_desc.h b/common/boot_desc.h
index a9b587f..a0f62fa 100644
--- a/common/boot_desc.h
+++ b/common/boot_desc.h
@@ -40,6 +40,9 @@
 	uint32_t img_size;
 	uint64_t loadaddr;
 	uint8_t chksum[SHA384_SUM_LEN];
+	uint8_t sec_ver;
+	uint32_t pub_key_size;
+	uint32_t sig_size;
 };
 
 /*
